### PR TITLE
feat: in docker debug support

### PR DESCRIPTION
**Pull request by dadrus**
There is no related issue. The PR adds the possibility to debug Kratos in Docker. Since my entire local setup is dockerized, I was using this each time I was working on a PR for Kratos. Since I think I'm not the only one having such a setup, I hope the community will find this option useful. If this PR is going to be accepted, I'll extend the documentation on how to make use of it.

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [x] I am following the [contributing code guidelines](../blob/master/CONTRIBUTING.md#contributing-code).
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security vulnerability. If this pull request addresses a security. vulnerability, I confirm that I got green light (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- ~~I have added tests that prove my fix is effective or that my feature works.~~ (No tests possible)
- [x] I have added or changed [the documentation](docs/docs).

### CODE REVIEW

**Comment by aeneasr** [comment]
Thank you! Maybe it makes sense to move this to `contrib`? On a side note, we usually delimit docker images with `-` not `.` :)

**Comment by dadrus** [comment]
You're welcome. :) Do you mean to move the corresponding code to `contrib` or to add the corresponding documentation there? Actually, I don't mind. I do however think the places where I put the two files are better suited for the purpose as these are easier to find for the people.
As for the documentation itself I was thinking about the `/docs/docs/debug` directory, but if you have a better place in mind, please let me know.

**Comment by dadrus** [comment]
@aeneasr : I've added documentation as well as a template file for docker-compose based setups. Please take a look.
Where can I reference the new document from, or is it automatically included?

File: docs/docs/debug/debugging_in_docker.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
+# Debugging Kratos in Docker
```
Comment (aeneasr) [inline docs/docs/debug/debugging_in_docker.md:1]:
```suggestion
---
id: debug-docker-delve-ory-kratos
title: Debugging Ory Kratos in Docker with Delve
---
```

**Review by aeneasr — COMMENTED** [review]
Thank you! The last change would be adding the docs to the sidebar :)

https://github.com/ory/kratos/blob/master/docs/sidebar.json

File: docs/docs/debug/debugging_in_docker.md (hunk 1 above)
Comment (dadrus) [reply docs/docs/debug/debugging_in_docker.md]: Done

**Comment by aeneasr** [comment]
🙏

**Pull request merged**